rust/kernel/io_mem.rs

```diff
@@ -177,6 +177,47 @@ impl<const SIZE: usize> IoMem<SIZE> {
         crate::build_assert!(Self::offset_ok::<T>(offset), "IoMem offset overflow");
     }
 
+    /// Copy memory block from an i/o memory into a buffer.
+    ///
+    /// # Examples
+    /// ```
+    /// # use kernel::prelude::*;
+    /// use kernel::io_mem::{self, IoMem, Resource};
+    ///
+    /// fn test(res: Resource) -> Result {
+    ///     // Create an i/o memory block of at least 100 bytes.
+    ///     let mem = unsafe { IoMem::<100>::try_new(res) }?;
+    ///
+    ///     let mut buffer: [i8; 32] = [0; 32];
+    ///
+    ///     // Memcpy 16 bytes from offset 10 of i/o memory block into the buffer.
+    ///     mem.try_memcpy_fromio(&mut buffer, 10, 16)?;
+    ///
+    ///     Ok(())
+    /// }
+    /// ```
+    pub fn try_memcpy_fromio(&self, from: &mut [i8], offset: usize, count: usize) -> Result {
+        if !IoMem::<SIZE>::offset_ok::<u8>(offset) {
+            return Err(Error::EINVAL);
+        }
+
+        // Check if memory block can be copied from i/o region
+        // and also whether it will fit into data buffer.
+        if count > from.len() || count > SIZE - offset {
+            return Err(Error::EINVAL);
+        }
```

Comment:
I just uploaded #683 -- please use `Self::offset_ok_of_val` from it instead of the two blocks above.

Reply:
Sure, thanks.